pyrsistent/_pclass.py

```diff
@@ -139,6 +139,10 @@ def transform(self, *transformations):
         return transform(self, transformations)
 
     def __eq__(self, other):
+        if self is other:
+            return True
+        if hash(self) != hash(other):
+            return False
```

Comment:
Some of those were from the other branch (which I've taken out of this branch).

Added tests for the other missing branches.